react-native-cli/index.js

```diff
@@ -20,43 +19,55 @@ var CLI_MODULE_PATH = function() {
   );
 };
 
-checkForVersionArgument();
-
 var cli;
-try {
-  cli = require(CLI_MODULE_PATH());
-} catch(e) {}
 
-if (cli) {
-  cli.run();
-} else {
+if (checkForVersionArgument()) {
+  try {
+    cli = require(CLI_MODULE_PATH());
+  } catch(e) {
+    processCommand();
+  }
+
+  if (cli) {
+    cli.run();
+  }
+}
+
+function processCommand() {
   var args = process.argv.slice(2);
-  if (args.length === 0) {
+  var command = args[0];
+
+  if (typeof command === 'undefined') {
     console.error(
       'You did not pass any commands, did you mean to run `react-native init`?'
     );
     process.exit(1);
-  }
-
-  switch (args[0]) {
-  case 'init':
-    if (args[1]) {
-      init(args[1]);
-    } else {
-      console.error(
-        'Usage: react-native init <ProjectName>'
-      );
-      process.exit(1);
-    }
-    break;
-  default:
+  } else if (command !== 'init') {
```

Comment:
I use `else if` here and a couple other places to make the code testable. `process.exit` gets mocked in the tests so execution does not terminate normally.